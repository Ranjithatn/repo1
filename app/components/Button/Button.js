import React, { Fragment } from "react";
import Icon from "../Icon/Icon";

const Button = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  className,
  disabled = false,
  id,
  type = "button",
  style={},
}) => {
  return (
    <button
      id={id}
      type={type}
      className={"button" + (className !== undefined ? " " + className : "")}
      disabled={disabled}
      onClick={onClick}
      style={ style }
    >
      {leftIcon && <Icon icon={leftIcon} />}
      { text &&<span>{text}</span> }
      {rightIcon && <Icon icon={rightIcon} />}
    </button>
  );
};

export default Button;
