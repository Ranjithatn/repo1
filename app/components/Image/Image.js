import React, { Fragment } from "react";

const Image = ({ className, id, alt, src, width, height, onClick, title, style={} }) => {
  return (
    <img
      onClick={onClick}
      id={id}
      title={title}
      className={"img" + (className !== undefined ? " " + className : "")}
      src={src}
      alt={alt}
      width={width}
      height={height}
      style={style}
    />
  );
};

export default Image;
