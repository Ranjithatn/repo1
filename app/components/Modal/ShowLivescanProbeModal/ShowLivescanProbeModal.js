import React, { Fragment } from "react";
import { connect } from "react-redux";

import Modal from "../Modal";
import Button from "../../Button/Button";
import Image from "../../Image/Image";
import { requestHideModal } from "../../../actions/modal";
import Textarea from "../../Textarea/Textarea";
import "./ShowLivescanProbeModal.scss";
import { LivescanProbeModalSelector} from "../../../selectors/liveScan";

export const ShowLivescanProbeModal = ({
  requestHideModal,
  formatMessage,
  requestCapture,
  modalData
}) => {
  const { ProbeData, jobType } = modalData;
  // console.log("modalData",modalData);
  return (
    <Modal
      title={ProbeData.position}
      // title={ jobType === 'Latent' ? formatMessage({id: "Unknown"}) : ProbeData.position}
      className="mugshot-modal"
      requestHideModal={requestHideModal}
      content={
        <Fragment>
          <div className="mugshot-center">
            <Image
              id="fingerImg"
              src={
                ProbeData
                  ? ProbeData.image.Base64Data
                    ? "data:image/png;base64," + ProbeData.image.Base64Data
                    : ProbeData.image
                      ? "data:image/png;base64," + ProbeData.image
                      : ""
                  : ""
              }
            />
          </div>
        </Fragment>
      }
      buttons={
        <Fragment>
          <div className="footer-content">

            <div className="right">
              <Button
                id="cancelBtn"
                className="is-primary"
                text={formatMessage({ id: "close" })}
                onClick={requestHideModal}

              />
            </div>
          </div>
        </Fragment>
      }
    />
  );
};

const mapState = state => ({

  modalData:LivescanProbeModalSelector(state)
});

export default connect(mapState, {
  requestHideModal
})(ShowLivescanProbeModal);

