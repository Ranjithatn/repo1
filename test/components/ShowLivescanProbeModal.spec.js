import React from "react";
import toJson from "enzyme-to-json";
import { ShowLivescanProbeModal } from "../../app/components/Modal/ShowLivescanProbeModal/ShowLivescanProbeModal";
import Button from "../../app/components/Button/Button";
import Image from "../../app/components/Image/Image";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let component, tree, instance;
let fingerImg, doneBtn;

let props = {
  requestHideModal: jest.fn(),
  modalData: { ProbeData: { position: "RightThumb", image: { Base64Data: "xyzz" } }, jobType: "" },
  formatMessage: jest.fn()
};

beforeEach(() => {
  component = mount(<ShowLivescanProbeModal {...props} />);
  fingerImg = component.find("img#fingerImg");
  doneBtn = component.find("button#cancelBtn");
  tree = toJson(component);
  instance = component.instance();
});

describe("<ShowLivescanProbeModal />", () => {
  it("should render without crashing", () => {
    expect(tree).toMatchSnapshot();
  });
//     it("should call requestInputFieldChanged when latent is toggled", () => {
//     let ProbeData: undefined;
//     latentToggle.simulate("change", event);
//     expect(props.requestInputFieldChanged).toHaveBeenCalled();
//   });
it('should have a image', () => {
    const output = component.find("img#fingerImg").length;
    expect(output).toEqual(1);
  })
it('should have a image', () => {
  const ProbeData= { position: "RightThumb", image: { Base64Data: "" } }; 
    const output = component.find("img#fingerImg").length;
    expect(output).toEqual(1);
  })
});
