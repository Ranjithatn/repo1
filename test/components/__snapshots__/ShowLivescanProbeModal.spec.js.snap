// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowLivescanProbeModal /> should render without crashing 1`] = `
<Component
  formatMessage={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "id": "close",
          },
        ],
      ],
    }
  }
  modalData={
    Object {
      "ProbeData": Object {
        "image": Object {
          "Base64Data": "xyzz",
        },
        "position": "RightThumb",
      },
      "jobType": "",
    }
  }
  requestHideModal={[MockFunction]}
>
  <Modal
    buttons={
      <UNDEFINED>
        <div
          className="footer-content"
        >
          <div
            className="right"
          >
            <Button
              className="is-primary"
              id="cancelBtn"
              onClick={[MockFunction]}
              text={undefined}
            />
          </div>
        </div>
      </UNDEFINED>
    }
    className="mugshot-modal"
    content={
      <UNDEFINED>
        <div
          className="mugshot-center"
        >
          <Image
            id="fingerImg"
            src="data:image/png;base64,xyzz"
          />
        </div>
      </UNDEFINED>
    }
    requestHideModal={[MockFunction]}
    title="RightThumb"
  >
    <div
      className="modal is-active mugshot-modal"
      style={
        Object {
          "zIndex": "50",
        }
      }
    >
      <div
        className="modal-background"
      />
      <div
        className="modal-card"
      >
        <header
          className="modal-card-head"
        >
          <p
            className="modal-card-title"
          >
            RightThumb
          </p>
          <button
            aria-label="close"
            className="delete is-large"
            name="btn-modal-close"
            onClick={[MockFunction]}
          />
        </header>
        <section
          className="modal-card-body"
        >
          <div
            className="mugshot-center"
          >
            <Image
              id="fingerImg"
              src="data:image/png;base64,xyzz"
            >
              <img
                className="img"
                id="fingerImg"
                src="data:image/png;base64,xyzz"
                style={Object {}}
              />
            </Image>
          </div>
        </section>
        <footer
          className="modal-card-foot"
        >
          <div
            className="footer-content"
          >
            <div
              className="right"
            >
              <Button
                className="is-primary"
                id="cancelBtn"
                onClick={[MockFunction]}
              >
                <button
                  className="button is-primary"
                  disabled={false}
                  id="cancelBtn"
                  onClick={[MockFunction]}
                  style={Object {}}
                  type="button"
                />
              </Button>
            </div>
          </div>
        </footer>
      </div>
    </div>
  </Modal>
</Component>
`;
